File: Backend/Data/ViewModels/Request/ResetPasswordEmailVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View model class for the resset password first step request. 
    /// This class only represents the first request that is to
    /// send the emial with the link to change the password
    /// </summary>
    public class ResetPasswordEmailVM
    {
        public string Email { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="email">User's email</param>
        public ResetPasswordEmailVM(string email)
        {
            this.Email = email;
        }
    }
}

File: Frontend/src/app/viewModels/requests/ResetPasswordEmailVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model for a reset password link request
 * This model its used to get a link to the specified email so that the user password can be changed
 */
export class ResetPasswordEmailVM implements toJSONString {

    /**
     * Construct's an instance of {@link ResetPasswordEmailVM}
     * 
     * @param email user's email.
     */
    constructor(
        public email: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify({ "email": this.email })
    }

}